### ISSUE TITLE
Review

### ISSUE THREAD

**Issue by oliviertassinari** [flatlogic/react-material-admin#1]
Hi guys, thank you for sharing this theme!

I have taken the time to look at the implementation, you can find my feedback in the following list:
- `mui-datatables` I was chatting with the author of this dependency, maybe you should consider `material-table`. We have started to promote them: https://next.material-ui.com/demos/tables/#material-table.
- `recompose` this package was [soft deprecated](https://github.com/acdlite/recompose#a-note-from-the-author-acdlite-oct-25-2018). With the growing React hooks trend, it will soon no longer be needed.
- We have solved a lot of issue in Material-UI v4. The beta version starts to be a good shape. You can consider upgrading. I hope it will solve the pain points you faced.
- In v4, tree sharing work, if you upgrade, you can do without the bundle size increase:
```diff
-import Typography from '@material-ui/core/Typography';
-import IconButton from '@material-ui/core/IconButton';
+import { IconButton, Typography } from '@material-ui/core';
```
- `tinycolor2` if you want to minimize bundle size, Material-UI has a colorManipular.js module. It's kind of private but we want to provide an official solution, we have expored using polished last week, we didn't move forward for bundle size concern.
- It's the first time I see the `package.json` in each folder approach, interesting.
- Did you consider using Prettier?
- You might want to set the component prop when using the Typography component to get the correct HTML semantic headline. e.g.
```diff
-<Typography component="span" variant="h5" color="primary" className={classes.textRow}>Oops. Looks like the page you're looking for no longer exists</Typography>
+<Typography variant="h5" color="primary" className={classes.textRow}>Oops. Looks like the page you're looking for no longer exists</Typography>
```
- `<Tab label="Login" classes={{ root: classes.tab }} />` I would probably just use the classsName, it's more idiomatic.

I hope it helps :)

**Comment by jeveloper** [comment]
@flatlogic So , no answer eh

**Comment by michaeldaineka** [comment]
Hi, Olivier!

Thank you for reviewing!

We've already replaced ```recompose``` by ```React Hooks```

```package.json``` had been added as an approach to get in touch with file by typing ```./components/App``` to prevent unnecessary adding ```App.js``` to the tip of the path.

Other points had been considered.

We're glad to offer you to take a look at [React Material Admin Full](https://flatlogic.com/templates/react-material-admin-full/demo). We think that our template can be a great part of your ```Material UI Marketplace```. 

Thanks, Flatlogic LLC Team
